packages/web3-core/src/web3_context.ts

```diff
@@ -72,21 +72,21 @@ export type Web3ContextInitOptions<
 
 export type Web3ContextConstructor<
 	// eslint-disable-next-line no-use-before-define, @typescript-eslint/no-explicit-any
-	T extends Web3Context<any>,
+	T extends Web3Context,
 	T2 extends unknown[],
 > = new (...args: [...extras: T2, context: Web3ContextObject]) => T;
 
 // To avoid circular dependencies, we need to export type from here.
 export type Web3ContextFactory<
 	// eslint-disable-next-line no-use-before-define, @typescript-eslint/no-explicit-any
-	T extends Web3Context<any>,
+	T extends Web3Context,
 	T2 extends unknown[],
 > = Web3ContextConstructor<T, T2> & {
 	fromContextObject(this: Web3ContextConstructor<T, T2>, contextObject: Web3ContextObject): T;
 };
 
 export class Web3Context<
-	API extends Web3APISpec,
+	API extends Web3APISpec = any,
```

Comment:
why not to default `Web3APISpec = unknown`? Instead of any.

Reply:
Good point!